app/receipt.go

```diff
@@ -0,0 +1,177 @@
+}
+
+func (app *App) AddCosmosEventsToEVMReceiptIfApplicable(ctx sdk.Context, tx sdk.Tx, checksum [32]byte, response abci.ResponseDeliverTx) {
+	if response.Code > 0 {
+		return
+	}
+	wasmEvents := GetEventsOfType(response, wasmtypes.WasmModuleEventType)
+	logs := []*ethtypes.Log{}
+	for _, wasmEvent := range wasmEvents {
+		contractAddr, found := GetAttributeValue(wasmEvent, wasmtypes.AttributeKeyContractAddr)
+		if !found {
+			continue
+		}
+		// check if there is a ERC20 pointer to contractAddr
+		pointerAddr, _, exists := app.EvmKeeper.GetERC20CW20Pointer(ctx, contractAddr)
+		if exists {
+			action, found := GetAttributeValue(wasmEvent, "action")
+			if !found {
+				continue
+			}
+			var topics []common.Hash
+			switch action {
+			case "mint", "burn", "send", "transfer", "transfer_from", "send_from", "burn_from":
+				topics = []common.Hash{
+					ERC20TransferTopic,
+					app.GetEvmAddressAttribute(ctx, wasmEvent, "from"),
+					app.GetEvmAddressAttribute(ctx, wasmEvent, "to"),
+				}
+				logs = append(logs, &ethtypes.Log{
+					Address: pointerAddr,
+					Index:   uint(len(logs)),
+					Topics:  topics,
+					Data:    common.BigToHash(GetAmountAttribute(wasmEvent)).Bytes(),
+				})
+			case "increase_allowance", "decrease_allowance":
+				ownerStr, found := GetAttributeValue(wasmEvent, "owner")
+				if !found {
+					continue
+				}
+				spenderStr, found := GetAttributeValue(wasmEvent, "spender")
+				if !found {
+					continue
+				}
+				topics := []common.Hash{
+					ERC20ApprovalTopic,
+					app.GetEvmAddressAttribute(ctx, wasmEvent, "owner"),
+					app.GetEvmAddressAttribute(ctx, wasmEvent, "spender"),
+				}
+				res, err := app.WasmKeeper.QuerySmart(
+					ctx,
+					sdk.MustAccAddressFromBech32(contractAddr),
+					[]byte(fmt.Sprintf("{\"allowance\":{\"owner\":\"%s\",\"spender\":\"%s\"}}", ownerStr, spenderStr)),
+				)
+				if err != nil {
+					continue
+				}
+				allowanceResponse := &AllowanceResponse{}
+				if err := json.Unmarshal(res, allowanceResponse); err != nil {
+					continue
+				}
+				logs = append(logs, &ethtypes.Log{
+					Address: pointerAddr,
+					Index:   uint(len(logs)),
+					Topics:  topics,
+					Data:    common.BigToHash(allowanceResponse.Allowance.BigInt()).Bytes(),
+				})
+			}
+			continue
+		}
+	}
+	if len(logs) == 0 {
+		return
+	}
+	txHash := common.BytesToHash(checksum[:])
+	if response.EvmTxInfo != nil {
+		txHash = common.HexToHash(response.EvmTxInfo.TxHash)
+	}
+	var bloom ethtypes.Bloom
+	if r, err := app.EvmKeeper.GetTransientReceipt(ctx, txHash); err == nil && r != nil {
+		r.Logs = append(r.Logs, utils.Map(logs, evmkeeper.ConvertEthLog)...)
+		bloom = ethtypes.CreateBloom(ethtypes.Receipts{&ethtypes.Receipt{Logs: evmkeeper.GetLogsForTx(r)}})
+		r.LogsBloom = bloom[:]
+		_ = app.EvmKeeper.SetTransientReceipt(ctx, txHash, r)
+	} else {
+		bloom = ethtypes.CreateBloom(ethtypes.Receipts{&ethtypes.Receipt{Logs: logs}})
+		receipt := &evmtypes.Receipt{
```

Comment:
is there some indicator or flag so that we can somehow optionally include/exclude these events from a result?